build/build.go

```diff
@@ -1612,3 +1619,40 @@ func noPrintFunc(opt map[string]Options) bool {
 	}
 	return true
 }
+
+func saveLocalState(so client.SolveOpt, opt Options, node builder.Node, configDir string) error {
+	var err error
+
+	if so.Ref == "" {
+		return nil
+	}
+
+	lp := opt.Inputs.ContextPath
+	dp := opt.Inputs.DockerfilePath
+	if lp != "" || dp != "" {
+		if lp != "" {
+			lp, err = filepath.Abs(lp)
+			if err != nil {
+				return err
+			}
+		}
+		if dp != "" {
+			dp, err = filepath.Abs(dp)
+			if err != nil {
+				return err
+			}
+		}
+		ls, err := localstate.New(configDir)
+		if err != nil {
+			return err
+		}
+		if err := ls.SaveRef(node.Builder, node.Name, so.Ref, localstate.State{
+			LocalPath:      lp,
+			DockerfilePath: dp,
```

Comment:
```suggestion
			LocalPath:      filepath.ToSlash(lp),
			DockerfilePath: filepath.ToSlash(dp),
```

Reply:
We do want this to be the actual path that can be looked up from the system. If we do `ToSlash` then it might not be possible to get back the original path.